IS2_velocity/correlation_processing.py

```diff
@@ -4,11 +4,352 @@
+                best_lag = lagvec[ix_peak]
+                best_shift = shift_vec[ix_peak]
+
+                #                                 ### ID peak correlation coefficient; 'raw' = no sub-sampling
+                #                                 plotting = False
+                #                                 best_lag, peak_corr_value = find_correlation_peak(lagvec, shift_vec, corr_normed, max_width, min_width, dx_interp, type, plotting)
+                #                                 best_shift = best_lag * dx
+
+                velocities[rgt][beam][xi] = best_shift / (dt / 365)
+                correlations[rgt][beam][xi] = corr_normed[ix_peak]
+                lags[rgt][beam][xi] = lagvec[ix_peak]
+
+
+            else:
+                ### If there are too many nans, just save a nan
+                velocities[rgt][beam][xi] = np.nan
+                correlations[rgt][beam][xi] = np.nan
+                lags[rgt][beam][xi] = np.nan
+
+        ### Output xy locations of midpoints as well
+        midpoints_xy[rgt][beam] = np.array(pyproj.Proj(3031)(midpoints_lons[rgt][beam], midpoints_lats[rgt][beam]))
+        #extract measures veloc outside of this loop
+
+        if saving:
+            measures_Vx_path = glob.glob( map_data_root + '*_Vx.tif')[0]
+            measures_Vy_path = glob.glob( map_data_root + '*_Vy.tif')[0]
+
+            ### Add velocities to hdf5 file for each beam
+            with h5py.File(h5_file_out, 'a') as f:
+                f[beam + '/x_atc'] = midpoints_x_atc[rgt][beam]  # assign x_atc value of half way along the segment
+                f[beam + '/latitudes'] = midpoints_lats[rgt][beam]  # assign x_atc value of half way along the segment
+                f[beam + '/longitudes'] = midpoints_lons[rgt][beam]  # assign x_atc value of half way along the segment
+                f[beam + '/velocities'] = velocities[rgt][beam]  # assign x_atc value of half way along the segment
+                f[beam + '/correlation_coefficients'] = correlations[rgt][beam]  # assign x_atc value of half way along the segment
+                f[beam + '/best_lags'] = lags[rgt][beam]  # assign x_atc value of half way along the segment
+                f[beam + '/segment_ids'] = midpoints_seg_ids[rgt][beam]
+                f[beam + '/first_cycle_time'] = str(Time(times[cycle1][beam][0]))
+                f[beam + '/second_cycle_time'] = str(Time(times[cycle2][beam][0]))
+                f[beam + '/Measures_v_along'] = get_measures_along_track_velocity(midpoints_xy[rgt][beam][0], midpoints_xy[rgt][beam][1], spatial_extent,
+                                                                          measures_Vx_path, measures_Vy_path)
+
+    return velocities, correlations, lags, midpoints_x_atc, midpoints_xy, midpoints_lons, midpoints_lats
+
+def velocity_old(x_in1, x_in2, dh1, dh2, dt, segment_length=2000, search_width=1000, along_track_step=100, dx=20, corr_threshold=.65):
```

Comment:
I am wondering whether this can be called calculate_velocity_single_beam() and then be called from the loop, just to clean things up a bit.